kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -224,4 +241,168 @@ static <O extends HasMetadata, R extends HasMetadata> Set<ResourceID> findReferr
+                                                                   List<Condition> conditions) {
+        var result = resourceSupplier.get();
+        result.setMetadata(new ObjectMetaBuilder()
+                .withName(ResourcesUtil.name(ingress))
+                .withNamespace(ResourcesUtil.namespace(ingress))
+                .withUid(ResourcesUtil.uid(ingress))
+                .build());
+        S status = statusSupplier.get();
+        conditionSetter.accept(status, conditions);
+        observedGenerationSetter.accept(status, ingress.getMetadata().getGeneration());
+        result.setStatus(status);
+        return result;
+    }
+
+    @NonNull
+    static KafkaService patchWithCondition(KafkaService service, Condition condition) {
+        return newStatus(
+                service,
+                KafkaService::new,
+                KafkaServiceStatus::new,
+                KafkaServiceStatus::setConditions,
+                KafkaServiceStatus::setObservedGeneration,
+                maybeAddOrUpdateCondition(
+                        Optional.of(service)
+                                .map(KafkaService::getStatus)
+                                .map(KafkaServiceStatus::getConditions)
+                                .orElse(List.of()),
+                        condition));
+    }
+
+    @NonNull
+    static KafkaProxyIngress patchWithCondition(KafkaProxyIngress ingress, Condition condition) {
+        return newStatus(
+                ingress,
+                KafkaProxyIngress::new,
+                KafkaProxyIngressStatus::new,
+                KafkaProxyIngressStatus::setConditions,
+                KafkaProxyIngressStatus::setObservedGeneration,
+                maybeAddOrUpdateCondition(
+                        Optional.of(ingress)
+                                .map(KafkaProxyIngress::getStatus)
+                                .map(KafkaProxyIngressStatus::getConditions)
+                                .orElse(List.of()),
+                        condition));
+    }
+
+    @NonNull
+    static KafkaProtocolFilter patchWithCondition(KafkaProtocolFilter filter, Condition condition) {
+        return newStatus(
+                filter,
+                KafkaProtocolFilter::new,
+                KafkaProtocolFilterStatus::new,
+                KafkaProtocolFilterStatus::setConditions,
+                KafkaProtocolFilterStatus::setObservedGeneration,
+                maybeAddOrUpdateCondition(
+                        Optional.of(filter)
+                                .map(KafkaProtocolFilter::getStatus)
+                                .map(KafkaProtocolFilterStatus::getConditions)
+                                .orElse(List.of()),
+                        condition));
+    }
+
+    static ConditionBuilder newConditionBuilder(Clock clock, HasMetadata observedGenerationSource) {
+        var now = ZonedDateTime.ofInstant(clock.instant(), ZoneId.of("Z"));
```

Comment:
This does mean that every condition is created with its own time down to some milli/nano second level. Is that what we want or do we want a single time that represents the reconciler pass?

Reply:
It does, but _right now_ each status only every has a single condition, so there's only a single `lastTransitionTime`. I think we can re-evaluate when we have a Reconciler which is setting two conditions in the same call to `reconcile()`.

Reply:
I find my self swinging both ways on this. 

1. If the intent is to have a single time across the reconciliation cycle then we should pass it in as an integral part of the API.
2. It will be generated exactly the same way at every calls site so we should inline it... 

I worry that if its kept down in this method we are likely to forget to re-visit it when we actually do have a reconciler with multiple conditions (but maybe we never do??).

Reply:
We're passing the `clock` around, so it's not like there's a call to `System.currentTimeMillis` lurking in there trying to be forgotten. I think the `clock` will be enough to remind us in that glorious future time of Reconcilers managing multiple Conditions.